pkg/sql/sem/tree/name_resolution_test.go

```diff
@@ -624,6 +625,7 @@ func TestResolveTablePatternOrName(t *testing.T) {
 		{`kv`, ``, mpath("public", "pg_catalog"), true, ``, ``, ``, `prefix or object not found`},
 		{`pg_tables`, ``, mpath("public", "pg_catalog"), true, `pg_tables`, `"".pg_catalog.pg_tables`, `.pg_catalog[0]`, ``},
 		{`pg_tables`, ``, mpath(), true, `pg_tables`, `"".pg_catalog.pg_tables`, `.pg_catalog[0]`, ``},
+		{`system.users`, ``, mpath(), true, `system.public.users`, `system.public.users`, `system.public[0]`, ``},
```

Comment:
system isn't a virtual table is it? should we also add a test with a virtual schema with no currdb set, or is this the wrong place for it? can we add a logic test also?

Reply:
yeah it's not virtual, so actually, this test i added works before and after my PR. Not a great test i suppose then :P 

(the reason i added it is because querying `system.users` with a curDB of "" is what the internal executor does, so just wanted to make sure I didn't break that.)

but yeah adding more vtable tests and logic tests is a good call

Reply:
actually L654-L658 seems like good tests. anything else in particular you want to test on that front? i'll still add logic tests